assets/js/components/dashboard-sharing/UserRoleSelect.js

```diff
@@ -50,191 +50,206 @@ const { useSelect, useDispatch } = Data;
 const ALL_CHIP_ID = 'all';
 const ALL_CHIP_DISPLAY_NAME = __( 'All', 'google-site-kit' );
 
-export default function UserRoleSelect( { moduleSlug, isLocked = false } ) {
-	const viewContext = useViewContext();
-	const wrapperRef = useRef();
-
-	const [ initialSharedRoles, setInitialSharedRoles ] = useState( [] );
-
-	const { setSharedRoles } = useDispatch( CORE_MODULES );
-	const { setValue } = useDispatch( CORE_UI );
-
-	const shareableRoles = useSelect( ( select ) =>
-		select( CORE_MODULES ).getShareableRoles()
-	);
-	const sharedRoles = useSelect( ( select ) =>
-		select( CORE_MODULES ).getSharedRoles( moduleSlug )
-	);
-	const editingUserRoleSelect = useSelect( ( select ) =>
-		select( CORE_UI ).getValue( EDITING_USER_ROLE_SELECT_SLUG_KEY )
-	);
-	const editMode = editingUserRoleSelect === moduleSlug;
-
-	useKeyCodesInside( [ ESCAPE ], wrapperRef, () => {
-		if ( editMode ) {
-			setValue( EDITING_USER_ROLE_SELECT_SLUG_KEY, undefined );
-		}
-	} );
-
-	const toggleEditMode = useCallback( () => {
-		if ( ! editMode ) {
-			if (
-				! isEqual(
-					[ ...( sharedRoles || [] ) ].sort(),
-					initialSharedRoles
-				)
-			) {
-				trackEvent(
-					`${ viewContext }_sharing`,
-					'change_shared_roles',
-					moduleSlug
-				);
+const UserRoleSelect = forwardRef(
+	( { moduleSlug, isLocked = false }, ref ) => {
+		const viewContext = useViewContext();
+		const roleSelectBtnRef = useRef();
```

Comment:
Let's always use full-word-variables whenever possible. It makes the code much more readable, and we can afford to store the extra bytes 😉 

(I can't make the suggestion because it's the whole file, but let's go with `roleSelectButtonRef`.)

Reply:
This has been updated, thank you @tofumatt!